core/src/com/mygdx/battleship/BattleshipBots/BattleshipBot.java

```diff
@@ -1,15 +1,36 @@
 package com.mygdx.battleship.BattleshipBots;
 
+import com.mygdx.battleship.BattleshipUtils;
 import com.mygdx.battleship.MoveResult;
 
-public interface BattleshipBot {
+import java.util.*;
 
-    String[][] getShipPlcements();
+abstract class BattleshipBot {
 
-    //A1....J10
-    String getMove();
+    protected String name; //name of bot
+    protected final int[] shipSizes = {2,3,3,4,5};
+    protected String[][] shipsLocations;
+    protected String move;
+    protected HashMap<String,String> botShots; //bot's shots
+    protected HashMap<String,String> opponentShots; //opponents shots
 
-    void setMyMoveResult(MoveResult result);
+	public BattleshipBot(String name) {
+        this.name = name;
+		shipsLocations = new String[5][0];
```

Comment:
Could you double check these dimensions? I believe you can do String[][5] to get an array that contains 5 String arrays of unspecified length.